### PR TITLE
[SPARK-7777][Streaming] Fix the flaky test in org.apache.spark.streaming.BasicOperationsSuite

### PR DESCRIPTION
**Pull request by zsxwing**
Just added a guard to make sure a batch has completed before moving to the next batch.

### CODE REVIEW

**Comment by zsxwing** [comment]
cc @tdas

File: streaming/src/test/scala/org/apache/spark/streaming/BasicOperationsSuite.scala [hunk 1]
```diff
@@ -587,7 +590,11 @@ class BasicOperationsSuite extends TestSuiteBase {
         for (i <- 0 until input.size) {
           testServer.send(input(i).toString + "\n")
           Thread.sleep(200)
+          val numCompletedBatches = batchCounter.getNumCompletedBatches
           clock.advance(batchDuration.milliseconds)
+          if (!batchCounter.waitUntilBatchesCompleted(numCompletedBatches + 1, 5000)) {
+            fail("A batch cannot complete in 5 seconds")
```
Comment (tdas) [inline streaming/src/test/scala/org/apache/spark/streaming/BasicOperationsSuite.scala]: "Batch took more that 5 seconds to complete"

**Comment by tdas** [comment]
LGTM. Were you able to reproduce the problem? And then this fixed it?

**Comment by zsxwing** [comment]
Cannot reproduce it... Just some guess

**Comment by andrewor14** [comment]
Nice! Great to see this being nixed.

**Comment by zsxwing** [comment]
retest this please

**Comment by tdas** [comment]
Merging this, since this is clearly more reliable than the code before.

**Comment by zsxwing** [comment]
@tdas this can be merged to branch-1.3 directly

**Comment by tdas** [comment]
All right . merging it then.

**Pull request closed without merging**